pkg/provider/remote/custom.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Nitric Pty Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v2"
+)
+
+// TODO: Move this into remote provider logic
+// A provider should be able to produce it's own template for a valid stack specification as stack specifications may change over time
```

Comment:
Can this be done before the PR is merged?

Reply:
This needs some work on the contracts for the remote providers, Probably some kind of service we can add the the proto contracts that allows us to output a template, but would need to be readable by the CLI, for user prompts/token replacement.